src/test/java/org/apache/datasketches/theta/UnionImplTest.java

```diff
@@ -54,14 +54,14 @@ public void checkUpdateWithSketch() {
     assertEquals(union.getResult().getEstimate(), k, 0.0);
   }
 
-  @Test(expectedExceptions = SketchesArgumentException.class)
-  public void checkCorruptedCompactFlag() {
+  @Test
+  public void checkUnorderedCompactFlag() {
     int k = 16;
     WritableMemory mem = WritableMemory.wrap(new byte[(k*8) + 24]);
     UpdateSketch sketch = Sketches.updateSketchBuilder().setNominalEntries(k).build();
     for (int i=0; i<k; i++) { sketch.update(i); }
     CompactSketch sketchInDirectOrd = sketch.compact(true, mem);
-    sketch.compact(false, mem); //corrupt memory
+    sketch.compact(false, mem); //change the order bit
```

Comment:
My you have a very good eye!  At first I thought you found a bug, but what you found was an inert test, which I have now improved, thanks!  

As of version 1.3.0, there was an additional layer of classes that were ordered and unordered subclasses of the HeapCompactSketch and the DirectCompactSketch. But this was really unnecessary. So I eliminated those 4 classes as they only differed by the ordered bit.  So before, there were two ways to determine whether the CompactSketch was ordered or not: the class type, and the ordered bit (memory) or boolean (on heap).  This test was designed to test the situation where the indicators were out-of-sync and make sure an error was thrown.  

With the new version there is only one indicator of order, either a boolean on heap or a bit set in Memory. 
As a result this test was inert.  So instead of removing it, I added additional checks to make sure that the union operation would successfully and correctly merge the same hashes whether the sketch order flag was set or not.

Reply:
It makes sense to remove the various implementations of the different variants.  When performing set operations between sets with different order bits, are the unordered sets sorted first internally?  I ask this because searching the internal hash keys should still work in theory, if the keys are unsorted.

Reply:
In an UpdateSketch the hashes are in a Knuth Open Address Double Hash (OADH) power-of-2 sized array; essentially a hash table, but a very efficient one.  The placement of hashes in a hash table is random, so the hashes are never in any order. The CompactSketch has all the same hashes, but all the empty slots of the hash table have been removed.  When creating the CS, the user has the option to have the hashes sorted or not. Sorting costs a little more time when creating the CS, but when merging CSs into a union, we can take advantage of "early-stop" which results in dramatic improvement in merge speed performance.  We never "search" the hashes in a CS. There is no need to do that.

The CS is the simplest representation of a sketch.  It contains primarily the hashes and Theta, that's it!  It is immutable and has no concept of "K" or "Nominal Entries".  Yet it can be used as input of all set operations.